apps/site/components/EOL/Modal.tsx

```diff
@@ -0,0 +1,156 @@
+  maxWidth?: string;
+}> = ({ vulnerabilities, maxWidth = 'max-w-2xs' }) => {
+  const t = useTranslations('components.eolModal');
+
+  return (
+    <table className="w-full">
+      <thead>
+        <tr>
+          <th>{t('table.cves')}</th>
+          <th>{t('table.severity')}</th>
+          <th>{t('table.overview')}</th>
+          <th>{t('table.details')}</th>
+        </tr>
+      </thead>
+      <tbody>
+        {vulnerabilities.map((vuln, i) => (
+          <tr key={i}>
+            <td>
+              {vuln.cve.length
+                ? vuln.cve.map(cveId => (
+                    <div key={cveId}>
+                      <LinkWithArrow
+                        href={`https://www.cve.org/CVERecord?id=${cveId}`}
+                        target="_blank"
+                        rel="noopener noreferrer"
+                      >
+                        {cveId}
+                      </LinkWithArrow>
+                    </div>
+                  ))
+                : '-'}
+            </td>
+            <td>
+              <VulnerabilityChip severity={vuln.severity} />
+            </td>
+            <td className={classNames(maxWidth, 'truncate')}>
+              {vuln.description || vuln.overview || '-'}
+            </td>
+            <td>
+              {vuln.ref ? (
+                <LinkWithArrow
+                  href={vuln.ref}
+                  target="_blank"
+                  rel="noopener noreferrer"
+                >
+                  {t('blogLinkText')}
+                </LinkWithArrow>
+              ) : (
+                '—'
+              )}
+            </td>
+          </tr>
+        ))}
+      </tbody>
+    </table>
+  );
+};
+
+const UnknownSeveritySection: FC<{
+  vulnerabilities: Array<Vulnerability>;
+  hasKnownVulns: boolean;
+}> = ({ vulnerabilities, hasKnownVulns }) => {
+  const t = useTranslations('components.eolModal');
+
+  if (!vulnerabilities.length) {
+    return null;
+  }
+
+  return (
+    <details open={!hasKnownVulns}>
+      <summary className="cursor-pointer font-semibold">
+        {t('showUnknownSeverities')} ({vulnerabilities.length})
+      </summary>
+      <div className="mt-4">
+        <VulnerabilitiesTable
+          vulnerabilities={vulnerabilities}
+          maxWidth={'max-w-3xs'}
+        />
+      </div>
+    </details>
+  );
+};
+
+const EOLModal: FC<ModalProps> = ({ open, closeModal, data }) => {
+  const { release, vulnerabilities } = data as EOLModalData;
+  const t = useTranslations('components.eolModal');
+
+  const modalHeading = t(release.codename ? 'title' : 'titleWithoutCodename', {
+    version: release.major,
+    codename: release.codename ?? '',
+  });
+
+  const [knownVulns, unknownVulns] = vulnerabilities.reduce(
+    (acc, vuln) => {
+      acc[vuln.severity === 'unknown' ? 1 : 0].push(vuln as KnownVulnerability);
+      return acc;
+    },
+    [[], []] as [Array<KnownVulnerability>, Array<Vulnerability>]
+  );
+
+  knownVulns.sort(
```

Comment:
Every time this component renders the sort method will be called, this seems unnecessary, no?